packages/tools/src/stateManagement/segmentation/SegmentationStateManager.ts

```diff
@@ -497,225 +566,153 @@ export default class SegmentationStateManager {
   }
 
   /**
-   * Retrieves an array of segmentation representations for a given viewport.
-   * @param viewportId - The ID of the viewport.
-   * @returns An array of SegmentationRepresentation objects.
+   * Removes a segmentation representation from the state.
+   * @param {string} viewportId - The ID of the viewport.
+   * @param {string} segmentationId - The ID of the segmentation.
+   * @param {SegmentationRepresentations} type - The type of segmentation representation.
    */
-  getSegmentationRepresentations(
-    viewportId: string
-  ): SegmentationRepresentation[] {
-    const viewport = this.state.viewports[viewportId];
-
-    if (!viewport) {
-      return [];
-    }
-
-    return Object.keys(viewport).map((segRepUID) => {
-      return this.getSegmentationRepresentation(segRepUID);
-    });
-  }
-
-  /**
-   * Removes a segmentation representation from the representations object.
-   * @param segmentationRepresentationUID - The UID of the segmentation representation to remove.
-   */
-  removeRepresentation(segmentationRepresentationUID: string): void {
-    delete this.state.representations[segmentationRepresentationUID];
-
-    // remove it from every viewports as well
-    Object.keys(this.state.viewports).forEach((viewportId) => {
-      delete this.state.viewports[viewportId][segmentationRepresentationUID];
-    });
-  }
-
-  /**
-   * Set the active segmentation representation for the give viewport
-   * @param viewportId - The Id of the tool group that owns the
-   * segmentation data.
-   * @param segmentationRepresentationUID - string
-   */
-  setActiveSegmentationRepresentation(
+  removeSegmentationRepresentation(
     viewportId: string,
-    segmentationRepresentationUID: string
+    segmentationId: string,
+    type: SegmentationRepresentations
   ): void {
-    Object.keys(this.state.viewports[viewportId]).forEach((segRepUID) => {
-      this.state.viewports[viewportId][segRepUID].active = false;
-    });
-
-    if (!this.state.viewports[viewportId]) {
-      this.state.viewports[viewportId] = {};
-    }
-
-    if (!this.state.viewports[viewportId][segmentationRepresentationUID]) {
-      this.state.viewports[viewportId][segmentationRepresentationUID] = {
-        active: false,
-        visible: true,
-        segmentsHidden: new Set(),
-      };
-    }
-
-    this.state.viewports[viewportId][segmentationRepresentationUID].active =
-      true;
-  }
+    const viewport = this.state.viewports[viewportId];
 
-  getActiveSegmentationRepresentation(
-    viewportId: string
-  ): SegmentationRepresentation | undefined {
-    if (!this.state.viewports?.[viewportId]) {
+    if (!viewport) {
       return;
     }
 
-    const activeSegRep = Object.entries(this.state.viewports[viewportId]).find(
-      ([, value]) => value.active
+    const viewportRendering = viewport.find(
+      (segRep) =>
+        segRep.segmentationId === segmentationId && segRep.type === type
```

Comment:
Can one omit the type here, to remove the first representation for that segmentationId?  It seems like the type probably isn't needed much of the time.

Reply:
hmm maybe, i'll keep it here, but not in the pulbic method, this is state manager

Reply:
I was just meaning make the find condition be:
segRep.segmentationId === segmentationId && (segRep.type === type || !type)